gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -487,8 +487,10 @@ private static void cleanupStagingDataPerTask(JobState jobState) {
     Map<String, ParallelRunner> parallelRunners = Maps.newHashMap();
     try {
       for (TaskState taskState : jobState.getTaskStates()) {
+        TaskState fatTaskState = new TaskState(taskState);
+        fatTaskState.addAllIfNotExist(jobState);
```

Comment:
Actually seems there's a bunch of config properties needed so it might be a good idea to keep the logic here instead of iterating through every possible config property. I think a better approach would be to put `fatTaskState.addAllIfNotExist(jobState);` into the `ParallelRunner`.